remote_version.go

```diff
@@ -0,0 +1,52 @@
+package main
+
+import (
+	"context"
+
+	"github.com/hansmi/paperhooks/pkg/client"
+	"github.com/prometheus/client_golang/prometheus"
+)
+
+type remoteVersionClient interface {
+	GetRemoteVersion(ctx context.Context) (*client.RemoteVersion, *client.Response, error)
+}
+
+type remoteVersionCollector struct {
+	cl remoteVersionClient
+
+	updateAvailableDesc *prometheus.Desc
+}
+
+func newRemoteVersionCollector(cl remoteVersionClient) *remoteVersionCollector {
+	return &remoteVersionCollector{
+		cl: cl,
+
+		updateAvailableDesc: prometheus.NewDesc("paperless_remote_version_update_available", "Whether an update is available.", []string{"version"}, nil),
+	}
+}
+
+func (c *remoteVersionCollector) describe(ch chan<- *prometheus.Desc) {
+	ch <- c.updateAvailableDesc
+}
+
+func (c *remoteVersionCollector) collect(ctx context.Context, ch chan<- prometheus.Metric) error {
+	remoteVersion, _, err := c.cl.GetRemoteVersion(ctx)
+	if err != nil {
+		return err
+	}
+
+	if len(remoteVersion.Version) == 0 {
+		return nil
+	}
+
+	var updateAvailable float64
+	if remoteVersion.UpdateAvailable {
+		updateAvailable = 1
+	} else {
+		updateAvailable = 0
```

Comment:
Zero is the, well, zero value for floats. There's no need for the `else` branch.

Reply:
Fixed in https://github.com/hansmi/prometheus-paperless-exporter/pull/57/commits/60d42b3db1f49e54fa9893148e51b3071d91123f